src/parser/ion_1_0/text/mod.rs

```diff
@@ -0,0 +1,2614 @@
+    map(
+        tuple((
+            take_hour_and_minute,
+            opt(preceded(char(COLON), take_second)),
+            take_offset,
+        )),
+        assemble_time,
+    )(i)
+}
+
+/// fragment
+/// OFFSET
+///     : 'Z'
+///     | PLUS_OR_MINUS HOUR ':' MINUTE
+///     ;
+fn take_offset(i: &str) -> IonResult<&str, UtcOffset> {
+    alt((
+        map(char('Z'), |_| UtcOffset::UTC),
+        map(
+            pair(take_plus_or_minus, take_hour_and_minute),
+            |(sign, (hour, minutes))| {
+                let minutes: i16 = ((hour as i16) * 60) + (minutes as i16);
+                let signed_minutes = if sign == '-' { -minutes } else { minutes };
+                UtcOffset::minutes(signed_minutes)
+            },
+        ),
+    ))(i)
+}
+
+/// fragment
+/// HOUR
+///     : [01] DEC_DIGIT
+///     | '2' [0-3]
+///     ;
+fn take_hour(i: &str) -> IonResult<&str, u32> {
+    map(
+        alt((
+            recognize(pair(one_of("01"), one_if(is_dec_digit))),
+            recognize(pair(char('2'), one_of("0123"))),
+        )),
+        |s: &str| {
+            s.parse::<u32>()
+                .expect("parser verified hour should be valid u32")
+        },
+    )(i)
+}
+
+/// fragment
+/// MINUTE
+///     : [0-5] DEC_DIGIT
+///     ;
+fn take_minute(i: &str) -> IonResult<&str, u32> {
+    map(
+        recognize(pair(one_of("012345"), one_if(is_dec_digit))),
+        |s: &str| {
+            s.parse::<u32>()
+                .expect("parser verified minute should be valid u32")
+        },
+    )(i)
+}
+
+/// note that W3C spec requires a digit after the '.'
+/// fragment
+/// SECOND
+///     : [0-5] DEC_DIGIT ('.' DEC_DIGIT+)?
+///     ;
+fn take_second(i: &str) -> IonResult<&str, (u32, Option<BigInt>)> {
+    let (i, s) = recognize(pair(one_of("012345"), one_if(is_dec_digit)))(i)?;
+    let (i, f) = opt(preceded(char('.'), take_while1(is_dec_digit)))(i)?;
+    let seconds = s
+        .parse::<u32>()
+        .expect("parser verified seconds should be valid u32");
+    if let Some(f) = f {
+        let fractional = str_to_bigint(f, 10)?;
+        Ok((i, (seconds, Some(fractional))))
+    } else {
+        Ok((i, (seconds, None)))
+    }
+}
+
+///
+/// Encoding Section: Ion Int
+///
+
+/// Helper for turning Vec<&str>s into BigInts. Or failing miserably.
+fn str_to_bigint<'a, T: AsRef<str>>(
+    digits: T,
+    radix: u32,
+) -> Result<BigInt, Err<IonError<&'a str>>> {
+    match BigInt::from_str_radix(digits.as_ref(), radix) {
+        Ok(bigint) => Ok(bigint),
+        Err(e) => Err(Err::Failure(IonError::from_format_error(
+            "",
+            FormatError::Text(TextFormatError::BigInt(e, digits.as_ref().to_string())),
+        ))),
+    }
+}
+
+/// Helper for turning Vec<&str>s into BigUints. Or failing miserably.
+fn str_vec_to_bigint(vec: Vec<&str>, radix: u32) -> Result<BigInt, Err<IonError<&str>>> {
+    let digits: String = vec.concat();
+    Ok(str_to_bigint(digits, radix)?)
+}
```

Comment:
This would also need to become a combinator, so that it can properly call `str_to_bigint` as a combinator